From 00d6b47db0a68d0bc91b252675a2165985426f5e Mon Sep 17 00:00:00 2001
Message-Id: <00d6b47db0a68d0bc91b252675a2165985426f5e.1574125554.git.liu.denton@gmail.com>
In-Reply-To: <cover.1574125554.git.liu.denton@gmail.com>
References: <cover.1574125554.git.liu.denton@gmail.com>
From: Denton Liu <liu.denton@gmail.com>
Date: Mon, 18 Nov 2019 16:06:30 -0800
Subject: [PATCH 2/3] t3206: demonstrate failure with notes
To: Git Mailing List <git@vger.kernel.org>
Cc: Johannes Schindelin <Johannes.Schindelin@gmx.de>,
    Thomas Gummerer <t.gummerer@gmail.com>

When a commit being range-diff'd has a note attached to it, the note
will be compared as well. However, this shouldn't happen since the
purpose of range-diff is to compare the difference between two
commit-ranges and, since the note attached to a commit is mutable, they
shouldn't be included as part of the comparison.

Demonstrate this failure in t3206.

Signed-off-by: Denton Liu <liu.denton@gmail.com>
---
 t/t3206-range-diff.sh | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/t/t3206-range-diff.sh b/t/t3206-range-diff.sh
index 87c6c029db..29b9a6f854 100755
--- a/t/t3206-range-diff.sh
+++ b/t/t3206-range-diff.sh
@@ -505,4 +505,19 @@ test_expect_success 'range-diff overrides diff.noprefix internally' '
 	git -c diff.noprefix=true range-diff HEAD^...
 '
 
+test_expect_failure 'range-diff does not compare notes' '
+	git notes add -m "topic note" topic &&
+	git notes add -m "unmodified note" unmodified &&
+	test_when_finished git notes remove topic unmodified &&
+	git range-diff --no-color master..topic master..unmodified \
+		>actual &&
+	cat >expected <<-EOF &&
+	1:  $(test_oid t1) = 1:  $(test_oid u1) s/5/A/
+	2:  $(test_oid t2) = 2:  $(test_oid u2) s/4/A/
+	3:  $(test_oid t3) = 3:  $(test_oid u3) s/11/B/
+	4:  $(test_oid t4) = 4:  $(test_oid u4) s/12/B/
+	EOF
+	test_cmp expected actual
+'
+
 test_done
-- 
2.24.0.420.g9ac4901264

